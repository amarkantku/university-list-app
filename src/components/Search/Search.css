
.search-box{
    width: 100%;
    position: relative;
    display: flex;
    height: 40px;
  }
.search-input{
    width: 100%;
    padding: 10px;
    border: 1px solid #1abc9c;
    border-right: none;
    outline: none;
    font-size: 16px;
    background: none;
}
.search-button{
    height: 40px;
    width: 40px;
    outline: none;
    cursor: pointer;
    border-left: none;
    border: 1px solid #1abc9c;
}

.svg-icon.search-icon {
    display: inline-block;
}
.svg-icon.search-icon:focus .search-path, .svg-icon.search-icon:hover .search-path {
    stroke: #299ecc;
}
.svg-icon.search-icon:active .search-path {
    stroke: #1abc9c;
    stroke-width: 2px;
}

  