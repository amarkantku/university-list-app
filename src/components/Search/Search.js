import React, { useState } from 'react';
import classNames from 'classnames/bind';
import styles from './Search.css';
import { ReactComponent as SearchSvg } from '../../svg/search.svg';
let clx = classNames.bind(styles);

export default function Search(props) {
    const [inputValue, setValue] = useState('');
    const onChange = (e) => {
        e.preventDefault();
        setValue(e.target.value)
    }
    const submit = (e) => {
        e.preventDefault();
        props.onClickSearch(inputValue);
        setValue('');
    }

    const handleKeyDown = (e) => {
        if (e.key === 'Enter' || e.keyCode === 13) {
            props.onClickSearch(inputValue);
            setValue('');
        }
    }
    return (
        <div data-testid='container' className={clx('container')}>
            <div data-testid='search-box' className={clx('search-box')}>
                <input data-testid='search-input' onKeyDown={handleKeyDown} onChange={onChange} value={inputValue} className={clx('search-input')} type="text" placeholder="Search.." />
                <button data-testid='search-button' onClick={submit} className={clx('search-button')}><SearchSvg /></button>
            </div>
        </div>
    )
}
