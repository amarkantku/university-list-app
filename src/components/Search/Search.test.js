import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Search from './Search';
const defaultProps = {
    onClickSearch: () => {}
}

describe('<Search />', () => {
    test('component should render', () => {
        const { asFragment } = render(<Search {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('should fire onKeyDown event', () => {
        const { getByTestId } = render(<Search {...defaultProps} />);
        const inputEle = getByTestId('search-input');
        expect(getByTestId('container')).toBeTruthy();
        expect(getByTestId('search-box')).toBeTruthy();
        expect(inputEle).toBeTruthy();
        fireEvent.keyPress(inputEle, { key: "Enter", keyCode: 13 });
    });

    test('should fire button click event', () => {
        const { getByTestId } = render(<Search {...defaultProps}/>);
        const buttonEle = getByTestId('search-button');
        expect(getByTestId('container')).toBeTruthy();
        expect(buttonEle).toBeTruthy();
        fireEvent.click(buttonEle);
    });
});