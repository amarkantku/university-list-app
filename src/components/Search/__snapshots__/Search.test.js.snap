// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should render 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="container"
  >
    <div
      class="search-box"
      data-testid="search-box"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search.."
        type="text"
        value=""
      />
      <button
        class="search-button"
        data-testid="search-button"
      >
        <svg>
          search.svg
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
